File: C++/Two_Sum.cpp
   fun twoSum(nums: IntArray, target: Int): IntArray {
        val hashMap = mutableMapOf() // value to index
        
        for((i,v) in nums.withIndex()) {
            val diff = target -v
            val indexForDiffValue = hashMap.get(diff)

            indexForDiffValue?.let {
                return@twoSum intArrayOf(indexForDiffValue, i)
            }
            
            hashMap[v] = i 
        }
        
        throw IllegalStateException("no solution found")
    }
}

File: Java/twoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int arr1[] = new int[2];
        for (int i=0;i<=nums.length-2;i++){
            for(int j=i+1;j<=nums.length-1;j++){
                if (target == nums[i]+nums[j])
                {
                    arr1[0]=i;
                    arr1[1]=j;
                }
        }
        }
        return arr1;
    }
}
